### DISCUSSION TITLE
Conditional Function Not Being Called Multiple Times in jsPsych Timeline

### DISCUSSION THREAD

**Discussion by nee1k** [jspsych/jsPsych#3198] [Q&A]
I'm currently working on a project using jsPsych, and I'm facing an issue with a conditional function within a timeline. Here's a simplified version of my code:

            var BlinkOneItem = {
                type: jsPsychHtmlKeyboardResponseNoerase,
                stimulus: jsPsych.timelineVariable('ItemBLANK'),
                choices: "NO_KEYS",
                trial_duration: 0,
            }
            var BlinkAllItems = {
                timeline: [ BlinkOneItem, WaitBlinkDuration, ShowOneItem, WaitInterBlinkDuration ],
                timeline_variables: ListOfBlinkingItems,
                conditional_function: function() {if(currentBlinkingItem++ < nBlinkingItems) {return true;} else {return false;}}
            }

In this setup, BlinkAllItems is intended to be a timeline where BlinkOneItem is repeated until nBlinkingItems is reached. However, the conditional function is behaving unexpectedly. It seems to be called only once, and subsequent checks are not being executed as expected. The value of currentBlinkingItem is supposed to be incremented within the condition, but it seems like the condition is evaluated only once.

The same code works in jsPsych 6.1. Any insights or suggestions on why the conditional function is not being called multiple times within the timeline would be greatly appreciated. Thank you!

**Comment by Shaobin-Jiang** [comment]
But `conditional_function` **IS** supposed to be called only once prior to the start of the timeline. The function you wish to be called multiple times is `loop_function`.
